packages/web/src/components/common/wallet-connector-menu/WalletConnectorMenu.tsx

```diff
@@ -115,9 +116,9 @@ const WalletConnectorMenu: React.FC<WalletConnectorMenuProps> = ({
   const menuRef = useRef<HTMLDivElement | null>(null);
 
   const balanceText = useMemo(() => {
-    const balance = gnotBalance || account?.balances?.[0].amount;
+    const balance = isLoadingGnotBalance ? account?.balances?.[0].amount : gnotBalance;
 
-    const formattedPrice = ((balance ?? 0) / (gnotToken?.decimals ?? 1)).toString().match(roundDownDecimalNumber(6))?.toString() ?? 0;
+    const formattedPrice = ((balance ?? 0) / (Math.pow(10, gnotToken?.decimals ?? 1))).toString().match(roundDownDecimalNumber(6))?.toString() ?? 0;
```

Comment:
This syntax can be expressed as a BigNumber.

`BigNumber(balance ?? 0).shiftedBy((gnotToken?.decimals ?? 1) * -1)`